crates/sui-framework/deps/move-stdlib/sources/type_name.move

```diff
@@ -24,6 +29,47 @@ module std::type_name {
         &self.name
     }
 
+    /// Get Address string (Base16 encoded), first part of the TypeName.
+    public fun get_address(self: &TypeName): String {
+        // Base16 (string) representation of an address has 2 symbols per byte.
+        let i = address::length() * 2;
+        let str_bytes = ascii::as_bytes(&self.name);
+        let addr_bytes = vector[];
+
+        // Reverse-read string from the last byte pushing the value
+        // into the end of `addr_bytes` vector.
+        while (i > 0) {
+            i = i - 1;
+            vector::push_back(
+                &mut addr_bytes,
+                *vector::borrow(str_bytes, i)
+            );
+        };
+
+        ascii::string(addr_bytes)
+    }
+
+    /// Get name of the module.
+    public fun get_module(self: &TypeName): String {
+        // Starts after address and a double colon: `<addr as HEX>::`
+        let i = address::length() * 2 + 2;
+        let str_bytes = ascii::as_bytes(&self.name);
+        let module_name = vector[];
+
+        loop {
+            let char = vector::borrow(str_bytes, i);
+            if (char != &ASCII_COLON) {
+                vector::push_back(&mut module_name, *char);
+                i = i + 1;
+            } else {
+                break
+            }
+        };
+
+        vector::reverse(&mut module_name);
```

Comment:
Maybe it's just late, and I'm being slow -- but I think this is also going to produce the reverse of the ordering that I expect.  If you had some tests for these new functions, it would help settle my mind that I am indeed being silly.

Reply:
Sorry, it's me being dumb. Ofc if we read byte per byte and push back it should work as expected.